src/parquet/column/properties.h

```diff
@@ -81,8 +82,11 @@ static int64_t DEFAULT_PAGE_SIZE = 1024 * 1024;
 static int64_t DEFAULT_DICTIONARY_PAGE_SIZE = DEFAULT_PAGE_SIZE;
 static bool DEFAULT_IS_DICTIONARY_ENABLED = true;
 static Encoding::type DEFAULT_ENCODING = Encoding::PLAIN;
+static Encoding::type DEFAULT_DICTIONARY_ENCODING = Encoding::PLAIN_DICTIONARY;
```

Comment:
This suggests that we should omit the dictionary options in https://github.com/apache/parquet-cpp/blob/master/src/parquet/types.h#L92 and add only a boolean flag to turn dictionary encoding on or off, like parquet-mr: https://github.com/apache/parquet-mr/blob/master/parquet-column/src/main/java/org/apache/parquet/column/ParquetProperties.java#L244

Then internally, we can use either `PLAIN_DICTIONARY` for both page types or `PLAIN` plus `RLE_DICTIONARY` depending on whether  we're writing V1 or V2 files. Exposing these internal details in the public API doesn't seem useful